tests/c_unittests/CMakeLists.txt

```diff
@@ -20,7 +20,7 @@
 # -fno-inline: just to be extra sure, no particular reason so far
 # -fno-builtin: GCC would optimize e.g. abs() and we would not be able to stub
 # -fno-stack-protector: https://stackoverflow.com/questions/12201625/disable-using-sprintf-chk
-# -D_FORTIFY_SOURCE=0: for Ubuntu with -O1+, https://stackoverflow.com/questions/34907503/gcc-fread-chk-warn-warning
+# -U_FORTIFY_SOURCE: for Ubuntu with -O1+, https://stackoverflow.com/questions/34907503/gcc-fread-chk-warn-warning
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CXX_STANDARD_FLAGS} -fno-inline -fno-builtin -fno-stack-protector -U_FORTIFY_SOURCE")
```

Comment:
It tries to do that, for the test binary only. It worked for the Ubuntu jobs here in the GHA, but it does not work for RHEL 7. Maybe because the `rpmbuild` adds its flags in some very smart way? The compile line on RHEL 7 is

> cd /mnt/build/BUILD/qpid-dispatch-1.19.0/tests/c_unittests && /usr/bin/c++   -D__STDC_NO_ATOMICS__ -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -std=c++11 -fno-inline -fno-builtin -fno-stack-protector -U_FORTIFY_SOURCE  -O2 -g -DNDEBUG -I/mnt/build/BUILD/qpid-dispatch-1.19.0/include -I/usr/include/python3.6m -I/mnt/build/BUILD/qpid-dispatch-1.19.0/src -I/mnt/build/BUILD/qpid-dispatch-1.19.0/tests/cpp-stub    -Wall -Wextra -Werror -Wpedantic -pedantic-errors -Wno-empty-body -Wno-implicit-fallthrough -Wno-unused-parameter -Wno-ignored-qualifiers -Wno-missing-field-initializers -Wno-sign-compare -Wno-type-limits -o CMakeFiles/c_unittests.dir/test_listener_startup.cpp.o -c /mnt/build/BUILD/qpid-dispatch-1.19.0/tests/c_unittests/test_listener_startup.cpp

Reply:
Don't you think that it is strange that the RHEL 7 compile line first does -D_FORTIFY_SOURCE=2 and then later on does -U_FORTIFY_SOURCE. It seems to me that it tries to first set and then nullify  _FORTIFY_SOURCE.

In any case, does any of the upstream CI builds use the -D_FORTIFY_SOURCE ?

Reply:
> Don't you think that it is strange that the RHEL 7 compile line first does -D_FORTIFY_SOURCE=2 and then later on does -U_FORTIFY_SOURCE.

That is the result of line 24 right here in `[ests/c_unittests/CMakeLists.txt`. What I find strange is that the _FORTIFY_SOURCE is actually not undefined by this. Maybe the `-Wp,` has some strange powers?

> In any case, does any of the upstream CI builds use the `-D_FORTIFY_SOURCE`?

Ubuntu in Travis CI, because Ubuntu enables _FORTIFY_SOURCE with -O1 or higher by default, at least according to https://stackoverflow.com/questions/34907503/gcc-fread-chk-warn-warning.

Reply:
ok sounds good. The only issue I have with the CMakeLists.txt code that nullifies _FORTIFY_SOURCE is that someone might explicitly set  -D_FORTIFY_SOURCE with the intention of not running these tests and it is being nullified by us and the tests will still run. Oh well, I think that is fine. Approved !

Reply:
while I think that setting `_FORTIFY_SOURCE` to disable tests is silly and such ideas should not be encouraged, you have a point saying that undefining options the way it's there now is not a good practice

thanks